packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -620,6 +629,7 @@ export class Bucket extends BucketBase {
       bucketArn: new cdk.Output(this, 'BucketArn', { value: this.bucketArn }).makeImportValue().toString(),
       bucketName: new cdk.Output(this, 'BucketName', { value: this.bucketName }).makeImportValue().toString(),
       bucketDomainName: new cdk.Output(this, 'DomainName', { value: this.domainName }).makeImportValue().toString(),
+      bucketWebsiteUrl: new cdk.Output(this, 'WebsiteURL', { value: this.bucketWebsiteUrl }).makeImportValue().toString()
```

Comment:
What happens if bucketWebsiteUrl is undefined?

Reply:
As in from a consumer's point of view? Currently it would just result in the property being `undefined` as you said.

I've had a go at creating separate test cases (for cases where a bucket does and doesn't have a website configuration) but I'm not 100% sure how to proceed. Looking at the other cases it seems the usual way of testing this stuff is to generate the CloudFormation and then assert against that. In this case, however, the value is just an `Fn::GetAtt` snippet:
```javascript
'exports the WebsiteURL is website configuration is enabled'(test: Test) {
  const stack = new cdk.Stack();
  const bucket = new s3.Bucket(stack, 'Website', {
    bucketName: 'my-bucket',
    websiteIndexDocument: 'index.html'
  });
  test.deepEqual(cdk.resolve(bucket.bucketWebsiteUrl), 'my-bucket-<REGION>.s3-website.amazonaws.com');
  test.done();
}
```

The are two problems I've ran into when attempting to create test cases like the above:
1. `AssertionError: { 'Fn::GetAtt': [ 'Website32962D0B', 'WebsiteURL' ] } deepEqual 'my-bucket-<REGION>.s3-website.amazonaws.com'`
    1. How can I compare against the _actual_ value as opposed to an `Fn::GetAtt` snippet? (I assumed calling `cdk.resolve` would have solved this
    1. How can I determine the region correctly? I can't see an example of how to do this in any of the other test cases

Reply:
> Currently it would just result in the property being undefined as you said.

why? if the user sets up website hosting for the bucket, I'd expect this to return the URL, no?

> How can I compare against the actual value as opposed to an Fn::GetAtt snippet? (I assumed calling cdk.resolve would have solved this

You can't. The "end result" of a CDK synth is a CloudFormation template. I'd expect the test to assert that the resulting template (or template part) is the right one (i.e. includes the GetAtt).

> How can I determine the region correctly? I can't see an example of how to do this in any of the other test cases
 
What I usually do is print out the output of `cdk.resolve`, inspect it to make sure it's what I expect, and then use it as my test expectation:

Reply:
> why? if the user sets up website hosting for the bucket, I'd expect this to return the URL, no?
Ah, I thought you were asking what would happen if the user _hadn't_ set up website hosting for the bucket. In the case where they _have_ set up website hosting for the bucket, `bucketWebsiteUrl` should never be `undefined`. Is there something you've spotted that might result it in being `undefined`?

> You can't. The "end result" of a CDK synth is a CloudFormation template. I'd expect the test to assert that the resulting template (or template part) is the right one (i.e. includes the GetAtt).
> How can I determine the region correctly? I can't see an example of how to do this in any of the other test cases
Gotcha, I'll give this a go. Thanks!